rmindepscheck.R

```diff
@@ -1,102 +1,10 @@
-
-  deps_refs <- vapply(
-    seq_len(nrow(deps)),
-    function(i) {
-      i_ref <- deps[i, "ref"]
-      i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-      i_pkg <- deps[i, "package"]
-      i_op <- deps[i, "op"]
-      i_ver <- deps[i, "version"]
-
-      # In case GH-type ref comes from Remotes field -> check if CRAN pkg first and if yes - overwrite ref type
-      if (is(i_ref_parsed, "remote_ref_github") && nrow(pkgcache::meta_cache_list(i_pkg))) {
-        i_ref <- sprintf("cran::%s", i_pkg)
-        i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-      }
-
-      # check if package from Bioconductor
-      if (is(i_ref_parsed, "remote_ref_standard")) {
-        i_pkg_cache <- pkgcache::meta_cache_list(i_pkg)
-        if (any(i_pkg_cache$mirror %in% pkgcache::bioc_repos())) {
-          i_ref <- gsub("^.*::", "bioc::",  i_ref_parsed$ref)
-          i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-        }
-      }
-
-      # resolve minimal version
-      if (is(i_ref_parsed, "remote_ref_standard") || is(i_ref_parsed, "remote_ref_cran")) {
-        i_pkg_cache <- pkgcache::meta_cache_list(i_pkg)
-        i_pkg_cache_archive <- pkgcache::cran_archive_list(package = i_pkg)
-        pv_all <- sort(package_version(unique(c(i_pkg_cache$version, i_pkg_cache_archive$version))), decreasing = TRUE)
-        pv_valid <- Filter(Negate(is.na), pv_all)
-        if (i_op != "") {
-          pv_valid <- Filter(function(x) do.call(i_op, list(x, package_version(i_ver))), pv_valid)
-        }
-        i_ver <- as.character(min(pv_valid))
-        i_ref <- sprintf("%s@%s", i_ref, i_ver)
-        i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-      } else if (is(i_ref_parsed, "remote_ref_github")) {
-        tags <- get_gh_tags(i_ref_parsed$username, i_ref_parsed$repo)
-        pv_all <- get_ver_from_tag(tags)
-        pv_valid <- Filter(Negate(is.na), pv_all)
-        if (i_op != "") {
-          pv_valid <- Filter(function(x) do.call(i_op, list(x, package_version(i_ver))), pv_valid)
-        }
-        tag_min <- tags[which(pv_all == min(pv_valid))]
-        i_ref <- sprintf("%s/%s@%s", i_ref_parsed$username, i_ref_parsed$repo, tag_min)
-        i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-      }
-
-      # @TODO: wait for https://github.com/r-lib/pak/issues/122
-      # as a suggested workaround - use GH mirror of CRAN
-      if (is(i_ref_parsed, "remote_ref_standard") || is(i_ref_parsed, "remote_ref_cran")) {
-        i_ref <- sprintf("cran/%s@%s", i_pkg, i_ver)
-        i_ref_parsed <- pkgdepends::parse_pkg_ref(i_ref)
-      }
-
-      return(i_ref)
-    },
-    character(1)
-  )
-
-  pkgdepends::new_pkg_installation_proposal(deps_refs, config = list(library = tempfile()))
-}
+catnl <- function(x) cat(sprintf("%s\n", x))
+catbr <- function() catnl("---")
+
+catbr()
+catnl("Install required packages")
+install.packages("remotes")
+remotes::install_github("insightsengineering/verdepcheck@*release")
```

Comment:
Not quite sure whether to use `@*release` or HEAD or let users decide by action parameter (with some default value). If the latter - shall we make use of `r-lib/actions/setup-r-dependencies@v2`?